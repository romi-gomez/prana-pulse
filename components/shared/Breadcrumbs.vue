<template>
  <nav aria-label="breadcrumb">
    <ol>
      <li v-for="(breadcrumb, index) in breadcrumbs" :key="index">
        <router-link :to="breadcrumb.href">{{ breadcrumb.label }}</router-link>
      </li>
    </ol>
  </nav>
</template>

<script setup>
import { defineProps } from 'vue'

const props = defineProps({
  breadcrumbs: {
    type: Array,
    required: true,
    default: () => []
  }
})
</script>

<style scoped>
/* Add any styles specific to this component */
</style>