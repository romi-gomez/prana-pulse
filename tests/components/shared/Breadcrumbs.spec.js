import { mount } from '@vue/test-utils'
import Breadcrumbs from '@/components/shared/Breadcrumbs.vue'
import { RouterLinkStub } from '@vue/test-utils'

describe('Breadcrumbs', () => {
  test('renders correctly', () => {
    const wrapper = mount(Breadcrumbs, {
      global: {
        stubs: {
          RouterLink: RouterLinkStub
        }
      },
      props: {
        breadcrumbs: [
          { label: 'Home', href: '/' },
          { label: 'Profile', href: '/profile' }
        ]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders breadcrumb links', () => {
    const wrapper = mount(Breadcrumbs, {
      global: {
        stubs: {
          RouterLink: RouterLinkStub
        }
      },
      props: {
        breadcrumbs: [
          { label: 'Home', href: '/' },
          { label: 'Profile', href: '/profile' }
        ]
      }
    })
    const links = wrapper.findAllComponents(RouterLinkStub)
    expect(links.length).toBe(2)
    expect(links[0].text()).toBe('Home')
    expect(links[1].text()).toBe('Profile')
  })
})