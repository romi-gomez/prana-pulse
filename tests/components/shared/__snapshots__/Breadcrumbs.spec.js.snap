// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders correctly 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol>
    
    <li>
      <a>
        Home
      </a>
    </li>
    <li>
      <a>
        Profile
      </a>
    </li>
    
  </ol>
</nav>
`;
